lnbits/extensions/boltz/views_api.py

```diff
@@ -0,0 +1,290 @@
+)
+async def api_mempool_url():
+    return BOLTZ_MEMPOOL_SPACE_URL
+
+
+# NORMAL SWAP
+@boltz_ext.get(
+    "/api/v1/swap",
+    name=f"boltz.get /swap",
+    summary="get a list of swaps a swap",
+    description="""
+        This endpoint gets a list of normal swaps.
+    """,
+    response_description="list of normal swaps",
+    dependencies=[Depends(get_key_type)],
+    response_model=List[SubmarineSwap],
+)
+async def api_submarineswap(
+    g: WalletTypeInfo = Depends(get_key_type),
+    all_wallets: bool = Query(False),
+):
+    wallet_ids = [g.wallet.id]
+    if all_wallets:
+        wallet_ids = (await get_user(g.wallet.user)).wallet_ids
+    return [swap.dict() for swap in await get_submarine_swaps(wallet_ids)]
+
+
+@boltz_ext.post(
+    "/api/v1/swap/refund",
+    name=f"boltz.swap_refund",
+    summary="refund of a swap",
+    description="""
+        This endpoint attempts to refund a normal swaps, creates onchain tx and sets swap status ro refunded.
+    """,
+    response_description="refunded swap with status set to refunded",
+    dependencies=[Depends(require_admin_key)],
+    response_model=SubmarineSwap,
+    responses={
+        400: {"description": "when swap_id is missing"},
+        404: {"description": "when swap is not found"},
+        405: {"description": "when swap is not pending"},
+        500: {
+            "description": "when something goes wrong creating the refund onchain tx"
+        },
+    },
+)
+async def api_submarineswap_refund(
+    swap_id: str,
+    g: WalletTypeInfo = Depends(require_admin_key),  # type: ignore
+):
+    if swap_id == None:
+        raise HTTPException(
+            status_code=HTTPStatus.BAD_REQUEST, detail="swap_id missing"
+        )
+
+    swap = await get_submarine_swap(swap_id)
+    if swap == None:
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND, detail="swap does not exist."
+        )
+
+    if swap.status != "pending":
+        raise HTTPException(
+            status_code=HTTPStatus.METHOD_NOT_ALLOWED, detail="swap is not pending."
+        )
+
+    try:
+        await create_refund_tx(swap)
+    except Exception as exc:
+        raise HTTPException(status_code=HTTPStatus.METHOD_NOT_ALLOWED, detail=str(exc))
+
+    await update_swap_status(swap.id, "refunded")
+    return swap
+
+
+@boltz_ext.post(
+    "/api/v1/swap",
+    status_code=status.HTTP_201_CREATED,
+    name=f"boltz.post /swap",
+    summary="create a submarine swap",
+    description="""
+        This endpoint creates a submarine swap
+    """,
+    response_description="create swap",
+    response_model=SubmarineSwap,
+    responses={
+        405: {"description": "not allowed method, insufficient balance"},
+        500: {"description": "boltz error"},
+    },
+)
+async def api_submarineswap_create(
+    data: CreateSubmarineSwap,
+    wallet: WalletTypeInfo = Depends(require_admin_key),  # type: ignore
+):
+    try:
+        data = await create_swap(data)
```

Comment:
You're overwriting `data` with a variable of a different type. Mypy didn't complain?

Reply:
no because, https://github.com/lnbits/lnbits-legend/pull/873, is not yet in :)

Reply:
changes it: 3f717ea3